core/src/test/java/org/hibernate/ogm/test/hint/SupportCustomHintTest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.test.hint;
+
+import static org.hibernate.ogm.test.hint.CustomHintSupportedProvider.Dialect.DIALECT_SPECIFIED_HINT;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+import javax.persistence.Query;
+
+import org.hibernate.ogm.test.datastore.Noise;
+import org.hibernate.ogm.utils.PackagingRule;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class SupportCustomHintTest {
```

Comment:
I worry that this test does not work without a subsequent commit. So maybe it is better to move this commit after or squash them in a single commit.